core/src/test/scala/unit/kafka/server/ListOffsetsRequestTest.scala

```diff
@@ -92,19 +93,30 @@ class ListOffsetsRequestTest extends BaseRequestTest {
     assertResponseError(Errors.NOT_LEADER_OR_FOLLOWER, nonReplica, debugReplicaRequest)
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testListOffsetsMaxTimeStampOldestVersion(quorum: String): Unit = {
+  @Test
+  def testListOffsetsMaxTimeStampOldestVersion(): Unit = {
     val consumerRequestBuilder = ListOffsetsRequest.Builder
-      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, false)
+      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, false, false)
 
     val maxTimestampRequestBuilder = ListOffsetsRequest.Builder
-      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, true)
+      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, true, false)
 
     assertEquals(0.toShort, consumerRequestBuilder.oldestAllowedVersion())
     assertEquals(7.toShort, maxTimestampRequestBuilder.oldestAllowedVersion())
   }
 
+  @Test
+  def testListOffsetsRequestTiredTimestampOldestVersion(): Unit = {
+    val consumerRequestBuilder = ListOffsetsRequest.Builder
+      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, false, false)
```

Comment:
This `consumerRequestBuilder` test is identical with the one in `testListOffsetsMaxTimeStampOldestVersion`. I think we can integrate these 2 tests into 1, and rename to something like: `testListOffsetsRequestOldestVersion` or something else. And I think (if we have time), we also need to test `requireTimestamp` and `isolationLevel == IsolationLevel.READ_COMMITTED` since I didn't see there's test covered them. WDYT?

Reply:
Yes, I add all cases to `testListOffsetsRequestOldestVersion`. Thanks.